### ISSUE TITLE
Exception

### ISSUE THREAD

**Issue by randydots** [AppsFlyerSDK/Unity#21]
When I use plugin, I encountered this exception when I called the  AppsFlyer.init() AppsFlyer.setAppID() for Android.

02-15 16:51:04.227: I/Unity(8457): 111111 AppsFlyer end in login ExceptionUnityEngine.AndroidJavaException: java.lang.RuntimeException: Can't create handler inside thread that has not called Looper.prepare()
02-15 16:51:04.227: I/Unity(8457):   at UnityEngine.AndroidJNISafe.CheckException () [0x00000] in <filename unknown>:0 
02-15 16:51:04.227: I/Unity(8457):   at UnityEngine.AndroidJNISafe.CallVoidMethod (IntPtr obj, IntPtr methodID, UnityEngine.jvalue[] args) [0x00000] in <filename unknown>:0 
02-15 16:51:04.227: I/Unity(8457):   at UnityEngine.AndroidJavaObject._Call (System.String methodName, System.Object[] args) [0x00000] in <filename unknown>:0 
02-15 16:51:04.227: I/Unity(8457):   at UnityEngine.AndroidJavaObject.Call (System.String methodName, System.Object[] args) [0x00000] in <filename unknown>:0 
02-15 16:51:04.227: I/Unity(8457):   at AppsFlyer.init (System.String key) [0x00000] in <filename unknown>:0 
02-15 16:51:04.227: I/Unity(8457):   at LoginManager.Start () [0x00000] in <filename unknown>:0 
02-15 16:51:04.227: I/Unity(8457):  
02-15 16:51:04.227: I/Unity(8457): (Filename: ./artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp Line: 64)

If I register the com.appsflyer.AppsFlyerOverrideActivity as` <category android:name="android.intent.category.LAUNCHER" />` in the manifest file as the example shows, it works.
But I can not do that for I have other activity register as `<category android:name="android.intent.category.LAUNCHER" />.`

In my AndroidManifest.xml, I only added 
```
<receiver android:name="com.appsflyer.MultipleInstallBroadcastReceiver" android:exported="true">
        <intent-filter>
          <action android:name="com.android.vending.INSTALL_REFERRER" />
        </intent-filter>
 </receiver>
 <meta-data android:name="AppsFlyerDevKey" android:value="YOUR_APPSFLYER_DEV_KEY_HERE"/>
```
Did I miss something? Any suggestions on this?

**Comment by randydots** [comment]
More information for reference

`02-15 16:32:49.931: I/Unity(5285): 111111 AppsFlyer end in login ExceptionUnityEngine.AndroidJavaException: java.lang.RuntimeException: Can't create handler inside thread that has not called Looper.prepare()
02-15 16:32:49.931: I/Unity(5285):   at UnityEngine.AndroidJNISafe.CheckException () [0x00000] in <filename unknown>:0 
02-15 16:32:49.931: I/Unity(5285):   at UnityEngine.AndroidJNISafe.CallVoidMethod (IntPtr obj, IntPtr methodID, UnityEngine.jvalue[] args) [0x00000] in <filename unknown>:0 
02-15 16:32:49.931: I/Unity(5285):   at UnityEngine.AndroidJavaObject._Call (System.String methodName, System.Object[] args) [0x00000] in <filename unknown>:0 
02-15 16:32:49.931: I/Unity(5285):   at UnityEngine.AndroidJavaObject.Call (System.String methodName, System.Object[] args) [0x00000] in <filename unknown>:0 
02-15 16:32:49.931: I/Unity(5285):   at AppsFlyer.init (System.String key) [0x00000] in <filename unknown>:0 
02-15 16:32:49.931: I/Unity(5285):   at LoginManager.Update () [0x00000] in <filename unknown>:0`

**Comment by randydots** [comment]
More information for reference.

02-15 16:32:49.931: I/Unity(5285): 111111 AppsFlyer end in login ExceptionUnityEngine.AndroidJavaException: java.lang.RuntimeException: Can't create handler inside thread that has not called Looper.prepare()
02-15 16:32:49.931: I/Unity(5285):   at UnityEngine.AndroidJNISafe.CheckException () [0x00000] in <filename unknown>:0 
02-15 16:32:49.931: I/Unity(5285):   at UnityEngine.AndroidJNISafe.CallVoidMethod (IntPtr obj, IntPtr methodID, UnityEngine.jvalue[] args) [0x00000] in <filename unknown>:0 
02-15 16:32:49.931: I/Unity(5285):   at UnityEngine.AndroidJavaObject._Call (System.String methodName, System.Object[] args) [0x00000] in <filename unknown>:0 
02-15 16:32:49.931: I/Unity(5285):   at UnityEngine.AndroidJavaObject.Call (System.String methodName, System.Object[] args) [0x00000] in <filename unknown>:0 
02-15 16:32:49.931: I/Unity(5285):   at AppsFlyer.init (System.String key) [0x00000] in <filename unknown>:0 
02-15 16:32:49.931: I/Unity(5285):   at LoginManager.Update () [0x00000] in <filename unknown>:0

**Comment by shilp-gupta** [comment]
I faced this issue as well. I integrated apps flyer without making any additions to the manifest.

It seems that the AppsFlyerLib.init (in the java world) requires that it be called from the UI thread. I achieved this by slightly modifying AppsFlyer.cs such that the AppsFlyer.init method doesn't directly call init in the java world, but switches threads (call "runOnUiThread" on the activity object and pass it a callback method) and then call init on AppsFlyerLib from the callback.

**Comment by randydots** [comment]
Hi shilp-gupta,

How did you do that?
Could you show me more demo?

**Comment by shilp-gupta** [comment]
@randydots 

I pushed my changes here https://github.com/shilp-gupta/AppsFlyerSDK-Unity/blob/master/Assets/Plugins/AppsFlyer.cs

Note that I haven't updated the .unitypackage or anything in the fork. I just updated the AppsFlyer.cs file with the patches I made to make it work.

**Comment by randydots** [comment]
@shilp-gupta 

Thanks buddy. It's solved. It really helps me out.

**Comment by randydots** [comment]
@shilp-gupta 

After a lot of test, I found the tracking event is successful not for 100% using the appflyer test APP.
Did this happen on your side?

**Comment by shilp-gupta** [comment]
@randydots From my experience (I have used several different tracking services in various apps), merely testing is only useful to confirm that things are functional. If you can see any data at all, go ahead and push them to production. Report events to another analytics service (I use game analytics) and compare.

**Comment by randydots** [comment]
@shilp-gupta 

Ok, I will use this code as temp solutions. And I will continue to contact AppsFlyer Team for more information. And also I will also develop a plugin of my own using the Primer31 Unity Activity sharing system https://gist.github.com/prime31/0908e6100d7e228f1add

**Comment by randydots** [comment]
@shilp-gupta 

I have got the feedback from the AppsFlyer team. I will use their old version SDK for the tracking.

**Comment by Sudarmin** [comment]
Hi, where do you get the old version SDK?

**Comment by maksimonishchenko** [comment]
ship gupta fix leading to https://github.com/AppsFlyerSDK/Unity/issues/22 in latestversion on this day. what old version is ok without manifest changes?

**Comment by golan19** [comment]
Fixed, please check release v4.3, Thanks everyone.
